HCCGo/app/js/fileManageService.js

```diff
@@ -317,6 +354,22 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
    }
    
    service.localHighlight = function(id) {
+      service.localOverwrite = false;
+      service.remoteOverwrite = false;
+      if(id.Class==="ext_txt") {
+        angular.element("#fileStats").show();
+        angular.element("#flocation").text(id.location);
+        angular.element("#fsize").text(id.size);
+        angular.element("#fmtime").text(id.mtime);
+      }
+      else {
+        angular.element("#fileStats").hide();
+      }
+      for(let dirObj of service.remoteFiles) {
+        if(id.name === dirObj.name) {
+          service.remoteOverwrite = true; 
+        }
+      }
```

Comment:
Eh, javascript and node are single threaded, a tight `for` loop is probably as good as it gets.